openedx/adg/admin_dashboard/admin_settings/tests/test_views.py

```diff
@@ -273,38 +261,200 @@ def test_base_list_view__edit_invitation(
         status = Invitation.PENDING
         email = invitation.email
 
-    request = request_factory.post(reverse(url), data={
-        'csrfmiddlewaretoken': 'csrfmiddlewaretoken',
-        'action_type': 'edit-invitation',
-        'email-0': email,
-        'permission-0': '2',
-        'business_unit-0': str(business_line.id)
-    })
+    request = request_factory.post(
+        reverse(url),
+        data=create_post_request_data(EDIT_INVITATION, email, Invitation.BU_ADMIN, business_line)
+    )
+    request.user = super_user
+    view.as_view()(request)
+
+    error = '' if object_exists else error.format(email=email)
+    invitations = create_invitations_dictionary(email, Invitation.BU_ADMIN, business_line, EDIT_INVITATION, status)
+    expected_context = create_base_list_view_context(key, url, error, invitations, EDIT_INVITATION)
+
+    mock_render.assert_called_once_with(request, f'adg/admin_dashboard/admin_settings/{template}', expected_context)
+
+
+@pytest.mark.django_db
+@skip_unless_lms
+@pytest.mark.parametrize(
+    'view, url, template, key, object_exists, error',
+    [
+        (
+            admin_dashboard_views.AdminAccountListView,
+            'admin_accounts',
+            'accounts.html',
+            'admins',
+            True,
+            ''
+        ),
+        (
+            admin_dashboard_views.AdminAccountListView,
+            'admin_accounts',
+            'accounts.html',
+            'admins',
+            False,
+            NO_ADMIN_ERROR
+        ),
+        (
+            admin_dashboard_views.PendingInvitationListView,
+            'pending_invitations',
+            'invitations.html',
+            'pending_invitations',
+            True,
+            ''
+        ),
+        (
+            admin_dashboard_views.PendingInvitationListView,
+            'pending_invitations',
+            'invitations.html',
+            'pending_invitations',
+            False,
+            NO_INVITATION_ERROR
+        ),
+    ],
+    ids=[
+        'admin_account_list_view_valid_post_request',
+        'admin_account_list_view_invalid_post_request',
+        'pending_invitation_list_view_valid_post_request',
+        'pending_invitation_list_view_invalid_post_request',
+    ]
+)
+def test_base_list_view__remove_invitation(
```

Comment:
test_base_list_view__remove_invitation is very similar to test_base_list_view__undo_remove_permissions & large chunk is being re-used. Can we please parametrize it?

Reply:
Mostly we use parametrisation in test cases for same scenarios but different inputs. But if scenario is different then we can separate it out. It will make our test case more readable and clean. The arguments that are being passed to the helper functions and factory are different hence, if we combine the two test cases the parameters would increase by a lot. The combined test case would be huge and difficult to understand. Also, if we use conditions instead of parametrisation that would also complicate the test case.